doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,352 @@
+automatically. The devices need to be able to:
+
+  - Sense temperature, light, humidity, smoke, etc.
+  - Integrate with home appliances to control temperature, light, power usage,
+    etc.
+  - Be usable but secure for people with no technical knowledge.
+  - Connect to a commercial home gateway.
+  - Communicate over common home IoT protocols such as ZigBee, Bluetooth, Wi-Fi,
+    etc.
+  - Connect to the building's power or provide their own power, depending on
+    the product.
+
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. This requires the board
+to:
+
+  - Perform its promised task reliably.
+  - Integrate and communicate reliably with the main processor.
+  - Include and support any relevant hardware.
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+This requires:
+
+  - The presence of didactic materials related to RIOT.
+  - Clear and easy to find documentation.
+  - A general availability of supported hardware and tools.
+
+# 3. Design philosophies
+
+Below are the design philosophies that are typically followed by developers to
+cater for the above use cases. The sections below include descriptions of
+tradeoffs between the philosophies, and where the resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't collecting data.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler. They shouldn't demand that users do
+the same.  They should, however, allow users to manage power through different
+modes and functions.
+
+#### Small memory footprint
+
+Apart from being optimized for low memory usage, RIOT tries very hard to be as
+modular as possible so unused features don't use up precious RAM or flash.
+Almost all features are provided as optional modules that have to be enabled
+explicitly at compile time.  A minimal RIOT configuration starts at around
+<2KiB flash and <1.5KiB RAM (including stack space for one thread and ISRs).
+
+Starting from there, the memory usage depends on the enabled features:
+
+- Non-networked control loop / sensing applications can fit on very small MCUs
+(eg., an Atmega328P with 2KiB RAM)
+- 6lowPAN networking currently starts at ~40KiB ROM and ~10KiB RAM
+- A 6lowpan enabled CoAP server requires ~60KiB ROM and ~15KiB RAM
+- a file system adds ~15 KiB ROM and ~2 KiB RAM
+
+#### Constrained networking
+
+RIOT should deliver best-in-class communication robustness and
```

Comment:
the implication is that the robustness we *should* provide is at least as good as anyone else's. To generalise somewhat, there are a number of statements in this document which reflect aspirations, rather than the actual current state. Hence the frequent use of the word "should". I believe this is appropriate and gives us goals to aim towards, which is an implicit purpose of a guidance document. (Guiding our developers to keep us exactly where we are would seem to be unproductive.)
I would say the question is more like: is "robustness as good as anyone else's" a specific enough, and realistic, goal?

Reply:
removed "best in class" as suggested